app/controllers/orders_controller.rb

```diff
@@ -1,47 +1,45 @@
 class OrdersController < ApplicationController
-  before_action :set_order, only: [:add_to_cart, :cart, :checkout, :update, :receipt]
+  before_action :set_order, only: [:cart, :checkout, :add_to_cart, :update, :receipt]
   before_action :set_product, only: [:add_to_cart]
 
-  def add_to_cart
-    if @order.already_has_product?(@product)
-      flash[:errors] = ERRORS[already_in_cart] # TODO: perhaps change this to incrementing the count in the cart?
+  def cart; end
+
+  def checkout; end
+
+  def add_to_cart # OPTIMIZE: consider moving this elsewhere, i.e. ProductsController or OrderItemsController.
+    order_item = OrderItem.new(product_id: @product.id, order_id: @order.id, quantity_ordered: 1)
+    if order_item.save
+      flash[:messages] = "The item has been added to your cart!"
     else
-      OrderItem.create(product_id: @product.id, order_id: @order.id, quantity_ordered: 1)
+      flash[:errors] = order_item.errors
     end
 
     redirect_to product_path(@product)
   end
 
-  def cart; end
-
-  def checkout; end
-
   def update
+    @order.status = "paid" # OPTIMIZE: it might make sense to make these be model methods.
     if @order.update(checkout_params)
       redirect_to receipt_path
     else
-      flash.now[:errors] = @order.errors
-      render :checkout
+      flash[:errors] = @order.errors
+      @order.status = "pending"
+      redirect_to checkout_path # TODO: might want to change this so that inputted data remains?
     end
   end
 
   def receipt
     if @order.status == "paid"
       render :receipt
-
-      reset_session # FIXME: this will interfere with login
+      session[:order_id] = nil
     else
-      redirect_to root_path # TODO: redirect to somewhere more logical
+      redirect_to root_path
```

Comment:
Because this is the only one that makes sense to me. `cart_path` is silly if they don't have an order, and there aren't really any other options.